sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -427,8 +427,8 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
   /**
    * All the top-level subqueries of the current plan node. Nested subqueries are not included.
    */
-  def subqueries: Seq[PlanType] = {
-    expressions.flatMap(_.collect {
+  lazy val subqueries: Seq[PlanType] = {
```

Comment:
please add `@transient`

Reply:
Thanks for tips, updated.

Reply:
Just for education purpose: why `@transient` is useful here?

Reply:
The `QueryPlan` e.g. `Expression` need to be sent to executor, use `@transient` to reduce the memory usage of executor.